DOC: add warning section in indexing docs.

zhangxiangnick opened the pull request:
Add warning section in indexing docs. Close #12947.

zhangxiangnick committed:
Add warning section in indexing docs. Close #12947.

jorisvandenbossche commented on doc/source/indexing.rst:
```diff
@@ -191,6 +191,17 @@ raised. Multiple columns can also be set in this manner:
+
+      df.loc[:,['B', 'A']] = df.loc[:,['A', 'B']]
+
+   ``df`` is not modified. The alignment is ignored in column assignment.
```
It would maybe be good to also give an illustration of the last sentence

jreback commented:
yeah this needs a bit more documentation of *why* this doesn't work and then what you should do instead.

zhangxiangnick committed:
Add warning section in indexing docs.

For assigning part of a DataFrame, explain why the assignment may not work and
then the correct way to do it.

zhangxiangnick commented:
Add why the DataFrame assignment may not work and then the correct to do it. Anything else I can do?

jreback commented on doc/source/indexing.rst:
```diff
@@ -191,6 +191,29 @@ raised. Multiple columns can also be set in this manner:
 
+.. warning::
+
+   pandas aligns index for setting a ``DataFrame`` from ``.loc``, ``.iloc`` and ``.ix``. 
```
aligns for when setting (not specific to ``DataFrame``)

jreback commented on doc/source/indexing.rst:
```diff
@@ -191,6 +191,29 @@ raised. Multiple columns can also be set in this manner:
+   .. code-block:: python
+
+      df.loc[:,['B', 'A']] = df.loc[:,['A', 'B']]
+
```
I would show the rhs here (as well)

zhangxiangnick replied:
What do you mean by "show here"?

jreback replied:
I would show ``df.loc[:, ['A', 'B']`` (e.g. what you are assigning)
